provider/pkg/await/awaiters.go

```diff
@@ -448,22 +415,13 @@ func untilCoreV1ResourceQuotaInitialized(c createAwaitConfig) error {
 		WatchUntil(rqInitialized, 1*time.Minute)
 }
 
-func untilCoreV1ResourceQuotaUpdated(c updateAwaitConfig) error {
-	oldSpec, _ := openapi.Pluck(c.lastOutputs.Object, "spec")
-	newSpec, _ := openapi.Pluck(c.currentOutputs.Object, "spec")
-	if !reflect.DeepEqual(oldSpec, newSpec) {
-		return untilCoreV1ResourceQuotaInitialized(c.createAwaitConfig)
-	}
```

Comment:
The old logic seems to say, if nothing has changed, then skip waiting for anything.  But waiting is harmless I suppose?

Reply:
Confirmed this still works as expected. Tested by adding an annotation to a ResourceQuota to exercise this case where the spec is unchanged -- previously we would have short-circuited but awaiting works just as well.